pcsx2/Gif.cpp

```diff
@@ -85,9 +192,33 @@ __fi void gifInterrupt()
 		return;
 	}	
 
+	if (CHECK_GIFFIFOHACK) {
+		
+		if (int amtRead = gif_fifo.read(true)) {
+
+			if (!gifUnit.Path3Masked() || gifRegs.stat.FQC < 16) {
+				GifDMAInt(amtRead * BIAS);
+				return;
+			}
+		}
+		else {
+
```

Comment:
That's only if the second statement is true. if it isn't it would fall through then try to match that other scenario, which it shouldn't do.